wagtail/admin/templates/wagtailadmin/pages/_unsaved_changes_warning.html

```diff
@@ -1,11 +1,32 @@
 {% load wagtailadmin_tags i18n %}
-<li class="footer__container footer__container--hidden" data-unsaved-warning>
-    <div hidden data-unsaved-type="any" class="footer__save-warning footer__emphasise-span-tags">
-        <div>
-            <b hidden data-unsaved-type="edits">{% blocktrans trimmed %}You have <span>unsaved edits</span>{% endblocktrans %}</b>
-            <b hidden data-unsaved-type="comments">{% blocktrans trimmed %}You have <span>unsaved comments</span>{% endblocktrans %}</b>
-            <b hidden data-unsaved-type="all">{% blocktrans trimmed %}You have <span>unsaved edits</span> and <span>comments</span>{% endblocktrans %}</b>
-            <p>{% trans 'Save the page before leaving' %}</p>
-        </div>
-    </div>
+<li
+    class="footer__container footer__container--hidden"
+    role="status"
+    data-controller="w-messages"
+    data-action="w-unsaved:add@document->w-messages#add w-unsaved:clear@document->w-messages#clear"
+    data-w-messages-clear-param="true"
+    data-w-messages-hide-class="footer__container--hidden"
+>
+    <div class="footer__save-warning footer__emphasise-span-tags" data-w-messages-target="container"></div>
+    {% trans 'Save the page before leaving' as generic_save_page_message %}
```

Comment:
Note: If we want, I can pull this 'shared' label up into the data-w-messages-text-param and remove from each of the `p` tags below. Not sure if that will make this HTML readability worse though.

Reply:
I don't think it's necessary, though perhaps it should be inside the `message_templates` block below, so if we later extract this into a shared template, we'll literally just have the block's contents in this file, e.g.

```html+django
{% extends "wagtailadmin/shared/unsaved_changes_warning.html" %}
{% block message_templates %}
    {% trans 'Save the page before leaving' as generic_save_page_message %}
    {# The rest of the templates here ... #}
{% endblock %}
```